NRLMMD-GEOIPS/geoips
# New docs

# Reviewer Checklist

* [x] NO REQUIRED ***existing tests*** (explain why not required)
* [x] NO REQUIRED ***updates to other repos***

https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/geoips#NNN

# Testing Instructions
- Build the geoips docs and inspect them
- Build docs for a plugin repo to verify that it still works

# Summary

- Removed "single_doc" functionality from docs builds because it was confusing and wasn't used by anyone
- Integrated new docs with old docs
- Split some documentation into multiple files / renamed docs where necessary
- Updated documentation categories to now be:
  - Getting Started
  - Tutorials
  - Concepts
      - Functionality
      - Architecture
  - Contribute
  - Releases
  - API 
  - Contact
- Updated the `Dockerfile` and `.devcontainer` files; updates to code no longer invalidates the pip install step cache

## Added Documentation about How to Add Documentation

![Screenshot 2024-12-04 at 5 53 05 PM](https://github.com/user-attachments/assets/848a52df-ac2e-4fea-850b-0d45c4880264)

![Screenshot 2024-12-04 at 5 53 14 PM](https://github.com/user-attachments/assets/9fbf7e07-460a-4177-9de0-3da2d96212ef)

![Screenshot 2024-12-04 at 6 13 01 PM](https://github.com/user-attachments/assets/fdc8b839-959e-4a3d-9963-455a9b43dec6)


# Output
View docs as built by the CI (under docs build step, click the link to download them) or build them yourself to view the new docs 🚀
